api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2680,3 +2689,49 @@ def test_extract_public_account_no_user_found(self, authenticated_client):
 
         response = self.post_to_endpoint(authenticated_client, user_id=42)
         assert response.status_code == 404
+
+
+class UnvalidatePublicAccountPasswordTest(PostEndpointHelper):
+    endpoint = "backoffice_web.public_accounts.invalidate_public_account_password"
+    endpoint_kwargs = {"user_id": 1}
+    needed_permission = perm_models.Permissions.MANAGE_PUBLIC_ACCOUNT
+
+    # session
+    # user
+    # targeted user by id
+    # validate user
+    # UPDATE user
+    # INSERT actionhistory
+    expected_queries = 6
+
+    def test_unvalidate_public_account_password(self, authenticated_client, legit_user):
+        user = users_factories.BeneficiaryFactory()
+        user_password_before_response = user.password
+
+        response = self.post_to_endpoint(
+            authenticated_client, user_id=user.id, expected_num_queries=self.expected_queries
+        )
+        assert response.status_code == 303
+
+        assert user_password_before_response != user.password
+
+        action_history = history_models.ActionHistory.query.one()
+        assert action_history.actionType == history_models.ActionType.USER_PASSWORD_INVALIDATED
+        assert action_history.authorUser == legit_user
+        assert action_history.user == user
+
+        response = authenticated_client.get(response.location)
+        assert "Le mot de passe du compte a bien été invalidé." in html_parser.extract_alert(response.data)
+
+    def test_unvalidate_public_account_password_user_not_found(self, authenticated_client):
+        response = self.post_to_endpoint(authenticated_client, user_id=42)
```

Comment:
```suggestion
        response = self.post_to_endpoint(authenticated_client, user_id=0)
```
l'id 42 pourrait exister dans la db, l'id 0 on est certain qu'il n'existe pas ;)